### PR TITLE
Fix for hosting code + unit tests. Fix #2340, Fix #2297,  Fix #2349

### LINKED ISSUE #2297: Rumor Mill under Street Peddler still affects corp cards
GitHub is having issues attaching the screenshot, but a player with Jackson Howard rezzed couldn't use him while a Street Peddler with Rumor Mill under it was in play. I'm guessing the same thing happens if Rumor Mill is installed face down via Apex or Hunting Grounds as well.

### LINKED ISSUE #2340: Street Peddler and Currents
If a current ends up on a Street Peddler, game thinks it's active.

**Rumor Mill** stopped Jackson from working.
**Hacktivist** caused trashes on asset-rez.

However, **Employee Strike** did _not_ appear to fire on asset-trash against CtM.

Things like Clot, The Source, etc. appear to be inactive, as expected.  So might just be isolated cases with some currents/events - I didn't do exhaustive research outside of the aforementioned cases.

### LINKED ISSUE #2349: Another Day Another Paycheck triggers indefinitely when not active
In a game today, Another Day Another Paycheck repeatedly triggered after an agenda steal, even though it was not in play as a current. After the first trace fired, it attempted to fire again, upon which we manually terminated it.

The runner stated that ADAP was installed on a Street Peddler, which may have been the source for the bug.

In addition, the runner also had severe trouble ending the run/jacking out afterwards

transcript below:

Reyne takes a mulligan.

Reyne
Good luck; have fun
idroj keeps their hand.

Reyne
don't see a ton of Sunny; hope it goes well :)
Reyne started their turn 1 with 5  and 5 cards in HQ.
Reyne spends 5  and  to play Hedge Fund.
Reyne uses Hedge Fund to gain 9 .
Reyne spends  to install ICE protecting HQ.
Reyne spends  to install ICE protecting R&D.
Reyne is ending their turn 1 with 9  and 3 cards in HQ.
idroj started their turn 1 with 5  and 5 cards in their Grip.
idroj spends 2  and  to play Dirty Laundry.
Reyne has no further action.
idroj uses Dirty Laundry to gain 5 .
idroj spends  and 1  to install Blockade Runner.
idroj spends  to use Blockade Runner to draw 3 cards and shuffle 1 card from their Grip back into their Stack.
idroj is ending their turn 1 with 7  and 5 cards in their Grip.
Reyne started their turn 2 with 9  and 3 cards in HQ.
Reyne spends  to install a card in Server 1 (new remote).
Reyne spends  to install a card in Server 2 (new remote).
Reyne spends  to gain 1 .
Reyne is ending their turn 2 with 10 and 2 cards in HQ.
idroj started their turn 2 with 7  and 5 cards in their Grip.
idroj spends 2  and  to play Dirty Laundry.
Reyne rezzes Jackson Howard.
Reyne removes Jackson Howard from the game.
Reyne uses Jackson Howard to shuffle Hedge Fund into R&D.

Reyne
not sure that was worth it
idroj spends  to make a run on Server 2.

Reyne
but extra spite points :P
Reyne has no further action.
idroj accesses Accelerated Beta Test from Server 2.
idroj steals Accelerated Beta Test and gains 2 agenda points.
idroj spends  to use Blockade Runner to draw 3 cards and shuffle 1 card from their Grip back into their Stack.
idroj trashes Power Tap from their Grip.
idroj is ending their turn 2 with 5  and 5 cards in their Grip.
Reyne started their turn 3 with 10  and 2 cards in HQ.
Reyne spends  to draw a card.
Reyne spends  to install ICE protecting Server 3 (new remote).
Reyne spends  to install a card in Server 3.
Reyne is ending their turn 3 with 10 and 2 cards in HQ.
idroj started their turn 3 with 5  and 5 cards in their Grip.
idroj spends  to make a run on Archives.
Reyne has no further action.
idroj spends  and 3  to install Daily Casts.
idroj spends  to install Data Leak Reversal.
idroj spends  to install Street Peddler.
idroj is ending their turn 3 with 2  and 2 cards in their Grip.
Reyne started their turn 4 with 10  and 2 cards in HQ.
Reyne spends  and 1  to advance a card in Server 3.
Reyne spends  and 1  to advance a card in Server 3.
Reyne spends  and 1  to advance a card in Server 3.
Reyne scores Project Vitruvius and gains 2 agenda points.
Reyne is ending their turn 4 with 7  and 3 cards in HQ.
idroj started their turn 4 with 2  and 2 cards in their Grip.
idroj uses Daily Casts to gain 2 .
idroj spends  to use Blockade Runner to draw 3 cards and shuffle 1 card from their Grip back into their Stack.
idroj uses Street Peddler to install Beth Kilrain-Chang lowering its install cost by 1 .
idroj spends 1  to install Beth Kilrain-Chang.
idroj spends  to gain 1 .
idroj spends  to gain 1 .
idroj is ending their turn 4 with 5  and 4 cards in their Grip.
Reyne started their turn 5 with 7  and 3 cards in HQ.
Reyne spends  to install ICE protecting Archives.
Reyne spends  to gain 1 .
Reyne spends  to install a card in Server 3.
Reyne is ending their turn 5 with 8  and 2 cards in HQ.
idroj started their turn 5 with 5  and 4 cards in their Grip.
idroj uses Beth Kilrain-Chang to gain 1 .
idroj uses Daily Casts to gain 2 .
idroj spends 5  and  to play Sure Gamble.
idroj uses Sure Gamble to gain 9 .
idroj spends  and 2  to install Citadel Sanctuary.
idroj spends  and 8  to install Security Nexus.
idroj spends  to draw a card.
idroj is ending their turn 5 with 2  and 2 cards in their Grip.
Reyne started their turn 6 with 8  and 2 cards in HQ.
Reyne spends  and 1  to advance a card in Server 3.
Reyne spends  and 1  to advance a card in Server 3.
Reyne spends  and 1  to advance a card in Server 3.
Reyne uses Accelerated Beta Test to look at the top 3 cards of R&D.

Reyne
eh, decent
Reyne installs Fairchild 3.0 protecting Server 3.
Reyne rezzes Fairchild 3.0 at no cost.
Reyne trashes a card.
Reyne trashes a card.
Reyne scores Accelerated Beta Test and gains 2 agenda points.
Reyne is ending their turn 6 with 5  and 3 cards in HQ.
idroj started their turn 6 with 2  and 2 cards in their Grip.
idroj uses Beth Kilrain-Chang to gain 1 .
idroj uses Daily Casts to gain 2 .
idroj spends  to make a run on Archives.
Reyne spends 3  to rez Ravana 1.0.
Reyne has no further action.
idroj spends  to break 1 subroutine on Ravana 1.0.
idroj spends  to break 1 subroutine on Ravana 1.0.
Reyne uses Haas-Bioroid: Architects of Tomorrow to rez Fairchild 3.0.
Reyne spends 2  to rez Fairchild 3.0.
idroj continues the run.
Reyne has no further action.
idroj spends  to draw a card.
idroj is ending their turn 6 with 5  and 3 cards in their Grip.
Reyne started their turn 7 with 0  and 3 cards in HQ.
Reyne spends  to gain 1 .
Reyne spends  to gain 1 .
Reyne spends  to install a card in Server 3.
Reyne is ending their turn 7 with 2  and 3 cards in HQ.
idroj started their turn 7 with 5  and 3 cards in their Grip.
idroj uses Daily Casts to gain 2 .
idroj spends  to make a run on Server 3.
idroj uses Security Nexus to force the Corp to initiate a trace.
Reyne uses Security Nexus to initiate a trace with strength 5 (5 + 0 ) (Give the Runner 1 tag and end the run).
idroj spends 0  to increase link strength to 3.
idroj uses Security Nexus to give the Runner 1 tag and end the run.
idroj spends  to use Data Leak Reversal to force the Corp to trash the top card of R&D.
idroj spends  to use Data Leak Reversal to force the Corp to trash the top card of R&D.
idroj spends  to use Data Leak Reversal to force the Corp to trash the top card of R&D.
idroj is ending their turn 7 with 7  and 3 cards in their Grip.
idroj uses Citadel Sanctuary to force the Corp to initiate a trace.
Reyne uses Citadel Sanctuary to initiate a trace with strength 1 (1 + 0 ) ().
idroj spends 0  to increase link strength to 3.
idroj uses Citadel Sanctuary to remove 1 tag.
Reyne started their turn 8 with 2  and 3 cards in HQ.
Reyne spends  to gain 1 .
Reyne spends  to gain 1 .
Reyne spends  to gain 1 .
Reyne is ending their turn 8 with 5  and 4 cards in HQ.
idroj started their turn 8 with 7  and 3 cards in their Grip.
idroj uses Beth Kilrain-Chang to gain 1 .
idroj spends  to make a run on Server 3.
idroj uses Security Nexus to force the Corp to initiate a trace.
Reyne uses Security Nexus to initiate a trace with strength 5 (5 + 0 ) (Give the Runner 1 tag and end the run).
idroj spends 0  to increase link strength to 3.
idroj uses Security Nexus to give the Runner 1 tag and end the run.
idroj spends  to use Data Leak Reversal to force the Corp to trash the top card of R&D.
idroj spends  to use Data Leak Reversal to force the Corp to trash the top card of R&D.
idroj spends  to use Data Leak Reversal to force the Corp to trash the top card of R&D.
idroj is ending their turn 8 with 8  and 3 cards in their Grip.
idroj uses Citadel Sanctuary to force the Corp to initiate a trace.
Reyne uses Citadel Sanctuary to initiate a trace with strength 1 (1 + 0 ) ().
idroj spends 0  to increase link strength to 3.
idroj uses Citadel Sanctuary to remove 1 tag.

Reyne
I need to run Successful Demonstration
Reyne started their turn 9 with 5  and 4 cards in HQ.
Reyne spends  to gain 1 .
Reyne spends  to gain 1 .
Reyne spends  to gain 1 .
Reyne is ending their turn 9 with 8  and 5 cards in HQ.
idroj started their turn 9 with 8  and 3 cards in their Grip.
idroj uses Beth Kilrain-Chang to gain 1 .
idroj spends  to make a run on Archives.
Reyne has no further action.
idroj uses Security Nexus to force the Corp to initiate a trace.
Reyne uses Security Nexus to initiate a trace with strength 5 (5 + 0 ) (Give the Runner 1 tag and end the run).
idroj spends 0  to increase link strength to 3.
idroj uses Security Nexus to give the Runner 1 tag and end the run.
idroj spends  to use Data Leak Reversal to force the Corp to trash the top card of R&D.
idroj spends  to use Data Leak Reversal to force the Corp to trash the top card of R&D.
idroj spends  to use Data Leak Reversal to force the Corp to trash the top card of R&D.
idroj is ending their turn 9 with 9  and 3 cards in their Grip.
idroj uses Citadel Sanctuary to force the Corp to initiate a trace.
Reyne uses Citadel Sanctuary to initiate a trace with strength 1 (1 + 0 ) ().
idroj spends 0  to increase link strength to 3.
idroj uses Citadel Sanctuary to remove 1 tag.
Reyne started their turn 10 with 8  and 5 cards in HQ.
Reyne spends  to gain 1 .
Reyne spends  to gain 1 .
Reyne spends  to gain 1 .
Reyne trashes a card from their HQ.
Reyne is ending their turn 10 with 11 and 5 cards in HQ.
idroj started their turn 10 with 9  and 3 cards in their Grip.
idroj uses Beth Kilrain-Chang to draw 1 card.
idroj spends  to make a run on Archives.
Reyne has no further action.
idroj spends  to break 1 subroutine on Ravana 1.0.
idroj spends  to break 1 subroutine on Ravana 1.0.
Reyne uses Haas-Bioroid: Architects of Tomorrow to rez Fairchild 3.0.
Reyne spends 2  to rez Fairchild 3.0.
idroj continues the run.
Reyne rezzes Jackson Howard.
Reyne removes Jackson Howard from the game.
Reyne uses Jackson Howard to shuffle 3 cards into R&D.
Reyne has no further action.
idroj spends  to install Street Peddler.
idroj is ending their turn 10 with 9  and 3 cards in their Grip.

Reyne
going to be so relieved when dlr rotates
Reyne started their turn 11 with 9  and 5 cards in HQ.
Reyne spends  to install a card in Server 3.
Reyne spends  to gain 1 .
Reyne spends  to gain 1 .
Reyne is ending their turn 11 with 11 and 5 cards in HQ.
idroj started their turn 11 with 9  and 3 cards in their Grip.
idroj uses Beth Kilrain-Chang to draw 1 card.
idroj spends  to make a run on Server 3.
idroj uses Security Nexus to force the Corp to initiate a trace.
Reyne uses Security Nexus to initiate a trace with strength 5 (5 + 0 ) (Give the Runner 1 tag and end the run).
idroj spends 2  to increase link strength to 5.
idroj uses Security Nexus to bypass the current ICE.
Reyne has no further action.
idroj continues the run.
Reyne has no further action.
idroj spends  to break 3 subroutines on Fairchild 3.0.
idroj continues the run.
Reyne has no further action.
idroj accesses Corporate Sales Team from Server 3.
idroj steals Corporate Sales Team and gains 2 agenda points.

Reyne
why trace?
Reyne uses Another Day, Another Paycheck to initiate a trace with strength 0 (0 + 0 ) ().

Reyne
that is not in effect

Reyne
bug?
idroj spends 0  to increase link strength to 3.
idroj uses Another Day, Another Paycheck to gain 8 .
!Reyne uses a command: /close-prompt
idroj sets credit to 14 (-1).
idroj sets credit to 13 (-1).
idroj sets credit to 12 (-1).
idroj sets credit to 11 (-1).
idroj sets credit to 10 (-1).
idroj sets credit to 9 (-1).
idroj sets credit to 8 (-1).
!Reyne uses a command: /close-prompt

Reyne
seriously

idroj
i in pelder

Reyne
ah

Reyne
I'll make an issue later in github?

idroj
y
!Reyne uses a command: /end-run
!Reyne uses a command: /close-prompt

Reyne
hrm, is it stuck for you?
!idroj uses a command: /end-run

### PR DESCRIPTION
**Pull request by danhut**
Fix #2340, Fix #2297,  Fix #2349

### CODE REVIEW

File: src/clj/game/core/hosting.clj [hunk 1]
```diff
@@ -71,7 +71,7 @@
 
        ;; events should be registered for: runner cards that are installed; corp cards that are Operations, or are installed and rezzed
        (when (or (is-type? target "Operation")
-                 (is-type? target "Event")
+                 (and (is-type? target "Event") (false? facedown))
```
Comment (nealterrell) [inline src/clj/game/core/hosting.clj]: Use `(not facedown)`, which allows `facedown` to be `nil`. (`(false? nil)` returns false.)

File: src/clj/test/cards/events.clj [hunk 2]
```diff
@@ -79,18 +79,25 @@
...
     (run-empty-server state :hq)
     (prompt-choice :runner "Steal")
     (prompt-choice :corp 0)
     (prompt-choice :runner 1)
-    ;; 4 credits after trace, should gain 4
-    (is (= 8 (:credit (get-runner))) "Runner gained 8 credits")))
+    ;; 4 credits after trace, gain 6
+    (is (= 10 (:credit (get-runner))) "Runner gained 6 credits")
+
+    ))
```
Comment (erbridge) [inline src/clj/test/cards/events.clj]: Just some stray brackets here :).

File: src/clj/game/core/hosting.clj (hunk 1 above)
Comment (erbridge) [inline src/clj/game/core/hosting.clj]: For future-proofing, shouldn't the `(not facedown)` requirement apply to both sides?

Comment (nealterrell) [inline src/clj/game/core/hosting.clj]: I can't see the game introducing "facedown" as an identical status for corp side; I imagine they'd name it something else. I think we're safe with assuming that keyword is for runner cards only.

Comment (danhut) [inline src/clj/game/core/hosting.clj]: That check on corp cards break at least casting call.  Maybe others - hence did not do it.  Thanks guys for reviews.  Damn bracket creep..!

**Pull request merged**